pcsx2/gui/MainFrame.cpp

```diff
@@ -765,24 +776,32 @@ void MainEmuFrame::CommitPreset_noTrigger()
 	g_Conf->EmuOptions.EnablePatches = menubar.IsChecked( MenuId_EnablePatches );
 }
 
-static void AppendShortcutToMenuOption( wxMenuItem& item, const char* id ) {
-	// this is NOT how a dictionary works but it has like 30 entries so this should still perform okay
-	auto* dict = &wxGetApp().GlobalAccels;
-	for ( auto it = ( *dict )->begin(); it != ( *dict )->end(); ++it ) {
-		if ( strcmp( it->second->Id, id ) == 0 ) {
-			wxString text = item.GetItemLabel();
-			size_t tabPos = text.rfind( L'\t' );
-			KeyAcceleratorCode keycode( (wxKeyCode)it->first );
-			item.SetItemLabel( text.Mid( 0, tabPos ) + L"\t" + keycode.ToString() );
-		}
-	}
+static void AppendShortcutToMenuOption( wxMenuItem& item, wxString keyCodeStr ) {
+	wxString text = item.GetItemLabel();
+	const size_t tabPos = text.rfind(L'\t');
+	item.SetItemLabel(text.Mid(0, tabPos ) + L"\t" + keyCodeStr);
 }
 
 void MainEmuFrame::AppendKeycodeNamesToMenuOptions() {
-	AppendShortcutToMenuOption( *m_menuSys.FindChildItem( MenuId_Sys_LoadStates ), "States_DefrostCurrentSlot" );
-	AppendShortcutToMenuOption( *m_menuSys.FindChildItem( MenuId_Sys_SaveStates ), "States_FreezeCurrentSlot" );
+	
+	AppendShortcutToMenuOption(*m_menuSys.FindChildItem( MenuId_Sys_LoadStates ), wxGetApp().GlobalAccels->findKeycodeWithCommandId("States_DefrostCurrentSlot").toTitleizedString());
+	AppendShortcutToMenuOption(*m_menuSys.FindChildItem( MenuId_Sys_SaveStates ), wxGetApp().GlobalAccels->findKeycodeWithCommandId("States_FreezeCurrentSlot").toTitleizedString());
 }
 
+#ifndef DISABLE_RECORDING
+void MainEmuFrame::initializeRecordingMenuItem(MenuIdentifiers menuId, wxString keyCodeStr) {
+	wxMenuItem& item = *m_menuRecording.FindChildItem(menuId);
+	wxString text = item.GetItemLabel();
+	const size_t tabPos = text.rfind(L'\t');
+	item.SetItemLabel(text.Mid(0, tabPos ) + L"\t" + keyCodeStr);
+	item.Enable(true);
+}
+void MainEmuFrame::enableRecordingMenuItem(MenuIdentifiers menuId, bool enable)
+{
+	wxMenuItem& item = *m_menuRecording.FindChildItem(menuId);
```

Comment:
The debugger threw a read access violation exception at this point - showing that `m_menuRecording.FindChildItem(menuId)` is already nullptr. This error will appear when you exit the emulator so long as you have booted an ISO at least one time - regardless of if recording tools are enabled or not.

Reply:
Ah ok, that might be due to the recent GUI refactor, I recall you finding an issue there as well?  I can solve that here at the same time...hopefully.  Looking into it.

Reply:
Should be resolved with the latest commit.  The `s*` macros automatically handle the `null` case for windows.